src/core/Ekubo.sol

```diff
@@ -0,0 +1,459 @@
+    function withdraw(address token, address recipient, uint128 amount) external;
+}
+
+IEkuboCore constant CORE = IEkuboCore(0xe0e0e08A6A4b9Dc7bD67BCB7aadE5cF48157d444);
+
+/// @notice Interface for the callback executed when an address locks core
+interface IEkuboCallbacks {
+    /// @notice Called by Core on `msg.sender` when a lock is acquired
+    /// @param id The id assigned to the action
+    /// @return Any data that you want to be returned from the lock call
+    function locked(uint256 id) external returns (bytes memory);
+
+    /// @notice Called by Core on `msg.sender` to collect assets
+    /// @param id The id assigned to the action
+    /// @param token The token to pay on
+    function payCallback(uint256 id, address token) external;
+}
+
+library UnsafeEkuboCore {
+    function unsafeSwap(
+        IEkuboCore core,
+        PoolKey memory poolKey,
+        int128 amount,
+        bool isToken1,
+        SqrtRatio sqrtRatioLimit,
+        uint256 skipAhead
+    ) internal returns (int128 delta0, int128 delta1) {
+        assembly ("memory-safe") {
+            let ptr := mload(0x40)
+
+            // mstore(ptr, 0x00000000) // selector for `swap_611415377((address,address,bytes32),int128,bool,uint96,uint256)`
+            // TODO: Check that native ETH is handled properly (0x0 vs 0xeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee)
+            mcopy(add(0x20, ptr), key, 0x60)
+            mstore(add(0x80, ptr), amount)
+            mstore(add(0xa0, ptr), isToken1)
+            mstore(add(0xc0, ptr), sqrtRatioLimit)
+            mstore(add(0xe0, ptr), skipAhead)
+            
+            if iszero(call(gas(), core, 0x00, add(0x1c, ptr), 0xe4, 0x00, 0x40)) {
+                returndatacopy(ptr, 0x00, returndatasize())
+                revert(ptr, returndatasize())
+            }
+            delta0 := mload(0x00)
+            delta1 := mload(0x20)
+        }
+    }
+
+    function unsafePay(IEkuboCore core, address token) internal returns (uint128 payment) {
+        assembly ("memory-safe") {
+            mstore(0x14, token)
+            mstore(0x00, 0x0c11dedd000000000000000000000000) // selector for `pay(address)` with `token`'s padding
+            if iszero(call(gas(), core, 0x00, 0x10, 0x24, 0x00, 0x20)) {
+                let ptr := mload(0x40)
+                returndatacopy(ptr, 0x00, returndatasize())
+                revert(ptr, returndatasize())
+            }
+            payment := mload(0x20)
+        }
+    }
+
+    function unsafeWithdraw(IEkuboCore core, address token, address recipient, uint128 amount) internal {
+        assembly ("memory-safe") {
+            let ptr := mload(0x40)
+
+            mstore(ptr, 0x03a65ab6) // selector for `withdraw(address,address,uint128)`
+            // TODO: Check that native ETH is handled properly (0x0 vs 0xeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee)
+            mstore(add(0x20, ptr), token)
+            mstore(add(0x40, ptr), recipient)
+            mstore(add(0x60, ptr), amount)
```

Comment:
All these values need to be masked/cleaned before putting them in memory. You can probably get a bit clever with this (see how Solady's SafeTransferLib handles `transferFrom`)

Reply:
Actually, does this function need to exist at all? If `Take.take` is working correctly, we should not need it